pkg/skaffold/build/options.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package build
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strconv"
+)
+
+// Configuration denotes build configuration for the artifact builder as either Release or Debug
+type Configuration int
+
+const (
+	Release = iota
+	Debug
+)
+
+// ImageOptions provides options for the artifact builder
+type ImageOptions struct {
+	// FIXME: Tag should be []string but we don't support multiple tags yet
+	Tag           string             // image tag
+	Configuration Configuration      // build image for release or debug
+	Args          map[string]*string // additional builder specific args
```

Comment:
I've renamed it to `BuildArgs`. 
The way I see this being used is in differentiating parameters that could influence build between the different skaffold actions like Dev, Build or Debug. Since skaffold.yaml doesn't have sections for these in case there are different settings required those can be put in this generic map. Whichever builder understands that setting would utilize it. This way since the setting affects the built image it also participates in cache hashcode calculation. WDYT?

Reply:
A concrete example could be to push the buildpacks builder image digest to this map so that if that changes it invalidates the cache. That'd solve this TODO:
![image](https://user-images.githubusercontent.com/39389231/87730841-c5b47980-c77d-11ea-9b2d-27889814d6bd.png)